arangod/Agency/AsyncAgencyComm.cpp

```diff
@@ -676,6 +677,38 @@ AsyncAgencyComm::FutureResult AsyncAgencyComm::deleteKey(
   return sendWriteTransaction(timeout, std::move(transaction));
 }
 
+AsyncAgencyComm::FutureResult AsyncAgencyComm::setTransientValue(
+    std::string const& key, velocypack::Slice const& slice,
+    network::Timeout timeout) {
+  VPackBuffer<uint8_t> transaction;
+  {
+    VPackBuilder trxBuilder(transaction);
+    VPackArrayBuilder env(&trxBuilder);
+    {
+      VPackArrayBuilder trx(&trxBuilder);
+      {
+        VPackObjectBuilder ops(&trxBuilder);
+        {
+          VPackObjectBuilder op(&trxBuilder, key);
+          trxBuilder.add("op", VPackValue("set"));
+          trxBuilder.add("new", slice);
+        }
+      }
+    }
+  }
+
+  RequestMeta meta;
+  meta.requestId = _manager.nextRequestId();
+  meta.method = RestVerb::Post;
+  meta.type = RequestType::CUSTOM;
+  meta.url = AGENCY_URL_TRANSIENT;
+  meta.timeout = timeout;
+  meta.skipScheduler = false;
```

Comment:
Do we want to use `skipScheduler = false` here? It seems to be the default value too, but do we really want to post the result on the scheduler for resolving?

Reply:
probably more work to post it, than actually executing the code 🤔 I will change it.

Reply:
wait a minute. Although this function is currently used in only one place, we don't know in general what is executed when the future is resolved. From that point of view I would argue for posting it on the scheduler.

Reply:
I see both points. How about the following: We could make `skipScheduler` an explicit argument of the method. Then one **has** to specify it at the call site and there one can judge if the attached `then` closures merit executing on the scheduler?